### ISSUE TITLE
[BUG] Some sound effects are getting pitched up (like 2.0 rate)

### ISSUE THREAD

**Issue by DarkBahamut162** [TeamRizu/OutFox#393]
**Please give System Info**

OS: Windows 10 Home 64-bit (21H1)
CPU: Intel Core i7 2600 @ 3.40GHz
GPU: ATI Radeon RX 580 Series 8GB
Storage: HDD

**Please give your StepMania Info**

StepMania Version: Project OutFox 5.3 Alpha 4.9.9HF3(b)
Theme: itg3 (the only theme where I've seen this happen, no matter if [Retropony's](https://github.com/Retropony/itg3-sm5-revival), [JoseVarelaP's](https://github.com/JoseVarelaP/itg3-sm5-revival) or [mine](https://github.com/DarkBahamut162/itg3encore))

**Describe the problem**

Certain sounds sound like they had the 2.0 rate pitch modifier being applied upon.
The only sounds with this problem so far:

ScreenCompany (immediately on StartTransitioning)
Screen cancel (after going further than ScreenTitleMenu, but resets after returning to Screen Company)
ScreenOptions next/prev (after going further than ScreenOptionsService, resets after returning to ScreenTitleMenu)

Started to happen after the HF3 upgrade (I streamed during HF2 and this problem wasn't there at that time).

**Describe what should happen**

Sounds should play out normally

**Logs and Additional context**

I don't think that logs matter here. But here is a [video](http://y2u.be/Y0gj48kJIWo)

**Comment by MrTwinkles47** [comment]
I have had a similar issue on 4.9.9HF3b with some announcer audio files. [Twitch Clip](https://clips.twitch.tv/GrossCreativeRadishChocolateRain-x6FCWeFCIh5s-G4r)
I have attached the audio file in the clip and also another one from earlier in the stream: 
[Announcer Clips.zip](https://github.com/TeamRizu/OutFox/files/6984141/Announcer.Clips.zip)
The files seem to both be 32kHz WAV files.

**Comment by aryoadhi** [comment]
Same issues happened, now with the chart song. Both the chart and the audio plays too fast in 4.9.9HF3, compared to previously I play on version 4.9.8GG.

The chart in question: Electric Dance System Music (from ECFA 2021 10s Day 1). [Link here.](https://drive.google.com/drive/folders/1213iPDGmrDeJ-yK74StkjH8c7linpipX)

Below is the output of `mediainfo` of problematic file, see if something bugs the game:

```
General
Complete name                            : Electric Dance System Music.ogg
Format                                   : Ogg
File size                                : 3.87 MiB
Duration                                 : 1 min 48 s
Overall bit rate mode                    : Variable
Overall bit rate                         : 299 kb/s
Album                                    : DanceDanceRevolution A [Analyzed Soundtrack]
Track name                               : Electric Dance System Music
Track name/Position                      : 2067
Performer                                : U1 overground
BPM (beats per minute)                   : 140
Length                                   : 113150
Band                                     : Various Artists

Audio
ID                                       : 22850 (0x5942)
Format                                   : Vorbis
Format settings, Floor                   : 1
Duration                                 : 1 min 48 s
Bit rate mode                            : Variable
Bit rate                                 : 300 kb/s
Channel(s)                               : 2 channels
Sampling rate                            : 32.0 kHz
Compression mode                         : Lossy
Stream size                              : 3.88 MiB
Writing library                          : libVorbis (Omnipresent) (20120203 (Omnipresent))
```

**Comment by DarkBahamut162** [comment]
I actually wanted to post something what I had found out about weird songs and what different Rate are making them do, but I stumbled myself into a rabbit hole:

All of this might be caused because of the **"Rate and Pitch Changes"** introduced in 4.9.9
The game pretty much expects EVERYTHING to be 44100Hz.
And I'm going to show you how wrong this is:

## Preview Oddities:
### **22050Hz|32000Hz**
Their preview in ScreenSelectMusic is already at Rate 2.0

## Haste Oddities:
### **44100Hz:**
Haste+Speed|Pitch: Does absolutely nothing!
Haste+Speed&Pitch: Acts like Haste+Speed

### **22050Hz|32000Hz|48000Hz:**
Haste+Speed: Only the stepchart has Haste while the song only goes down in pitch and stays at normal speed
Haste+Pitch: Same as above but even worse
Haste+Speed&Pitch: Same as above but even worse

## Speed Oddities:
### **22050Hz|32000Hz|48000Hz:**
**Speed+Rate0.5**
Sound: Like it went to Rate 2.0 and back to Rate 1.0 but mangled/bitcrushed (?) and gradually gets pitched up to Rate 2.0 (like its on Haste)
Chart: Starts on Rate 0.5 but gets reverse Haste (!), reversing if BPMs are slow enough. After a short while both the chart gets itself back together but staying at Rate 0.5 (no Haste compared to the audio)

**Speed+Rate1.0**
Sound&Stepchart: Both play and sound like it's on Rate 2.0

**Speed+Rate2.0**
Sound: Sounds like being played on Rate 2.0 but the pitch goes down in pitch (like a reverse Haste)
Chart: Seems like its even higher than Rate 2.0, maybe since it's gained Haste as well? WORSE ON "SPEED & PITCH"

### **44100Hz:**
Everything works as intended (except with Haste)

___

The problem seems to be the song's **SampleRate**, as BitsPerSample, Bitrate, Codecs or FileExtension didn't seem to have an affect on this.
And during all of this: The AssistTick seems to be somewhat delayed.
Also: Everything gets worse the farther away a song is from 44100Hz.

**Comment by SheepyChris** [comment]
The issue has been identified and fixed in preview/testing builds, will be closed when the hotfix is released. For the time being, enable the old resampler system in Preferences.ini

**Comment by Scraticus** [comment]
Can confirmed it will be released in HF4 which is due in the next few days.

**Comment by Scraticus** [comment]
Fixed in 4.9.10 (HF4)
